packages/metro/src/DeltaBundler/traverseDependencies.js

```diff
@@ -471,18 +475,47 @@ function resolveDependencies<T>(
   dependencies: $ReadOnlyArray<TransformResultDependency>,
   options: InternalOptions<T>,
 ): Map<string, Dependency> {
-  return new Map(
-    dependencies.map((result: TransformResultDependency) => {
-      const relativePath = result.name;
-
-      const dependency = {
-        absolutePath: options.resolve(parentPath, result.name),
-        data: result,
-      };
+  const {allowOptionalDependencies} = options;
+
+  const _resolve = (parentPath: string, result: TransformResultDependency) => {
+    const relativePath = result.name;
+    try {
+      return [
+        relativePath,
+        {
+          absolutePath: options.resolve(parentPath, relativePath),
+          data: result,
+        },
+      ];
+    } catch (e) {
+      // if this is an optional dependency, ignore the resolve error
+      if (
+        result.data.isOptional === true &&
+        allowOptionalDependencies !== false &&
+        !(
+          Array.isArray(allowOptionalDependencies.exclude) &&
+          allowOptionalDependencies.exclude.includes(relativePath)
```

Comment:
Should we instead invert this and require users to explicitly include the optional dependencies they want to use? That way we ensure users will update a configuration every time they use an optional dependency and that they will never end up with a broken bundle that is missing a dependency accidentally.

Reply:
maybe I didn't read your comment correctly... but that is exactly "allowOptionalDependencies.exclude" does: to turn optional dependency to be non-optional, if one chooses to.

But it's not required, because many 3rd-party library users might not know what is considered optional within those libraries... for those use cases, metro should just work (with standard syntax analysis - babel parser - taking the hints from code) without requiring extra user config, IMHO.

did I miss your point?